File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/subjects/CompletableSubjectExamples.java
package com.egecius.rxjava2_demo_2.rx.subjects;

import com.egecius.rxjava2_demo_2.rx.EgisException;

import io.reactivex.subjects.CompletableSubject;

public class CompletableSubjectExamples {

    private CompletableSubject mCompletableSubject = CompletableSubject.create();

    public CompletableSubject getCompletableSubject() {
        return mCompletableSubject;
    }

    void emit2Errors() {
        mCompletableSubject.onError(new EgisException());
        mCompletableSubject.onError(new EgisException());
    }

    void emit1Error() {
        mCompletableSubject.onError(new EgisException());
    }
}

File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/subjects/CompletableSubjectExamples.kt
package com.egecius.rxjava2_demo_2.rx.subjects

import com.egecius.rxjava2_demo_2.rx.EgisException

import io.reactivex.Completable
import io.reactivex.subjects.CompletableSubject

class CompletableSubjectExamples {

    val completableSubject = CompletableSubject.create()

    fun emit1Error() {
        completableSubject.onError(EgisException())
    }

    fun emit2Errors() {
        completableSubject.onError(EgisException())
        completableSubject.onError(EgisException())
    }

    fun subscribeToAnother() {
        val completableToCompleteImmediately = Completable.error(EgisException())

        completableToCompleteImmediately.subscribe(completableSubject)
    }
}
